docs/vcpkg-schema-definitions.schema.json

```diff
@@ -0,0 +1,715 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema",
+  "$id": "https://raw.githubusercontent.com/microsoft/vcpkg-tool/master/docs/vcpkg-schema-definitions.schema.json",
+  "definitions": {
+    "artifact-references": {
+      "description": "A key/value pair of artifact name to selected artifact version.",
+      "type": "object",
+      "minProperties": 1,
+      "patternProperties": {
+        "": {
+          "type": "string"
+        }
+      }
+    },
+    "artifact-registry": {
+      "description": "An artifact registry backed by a git repository.",
+      "type": "object",
+      "properties": {
+        "kind": {
+          "type": "string",
+          "pattern": "artifact"
+        },
+        "name": {
+          "$ref": "#/definitions/identifier"
+        },
+        "location": {
+          "type": "string",
+          "format": "uri",
+          "description": "A URI to the location of the artifact registry; typically a git URI."
+        }
+      },
+      "required": [
+        "kind",
+        "name",
+        "location"
+      ]
+    },
+    "builtin-registry": {
+      "type": "object",
+      "properties": {
+        "baseline": {
+          "$ref": "#/definitions/sha1",
+          "description": "The SHA that will be checked out to find default versions for ports from this registry."
+        },
+        "kind": {
+          "type": "string",
+          "pattern": "builtin"
+        }
+      },
+      "required": [
+        "baseline",
+        "kind"
+      ]
+    },
+    "contact": {
+      "type": "object",
+      "properties": {
+        "email": {
+          "type": "string"
+        },
+        "role": {
+          "$ref": "#/definitions/string-or-strings"
+        }
+      },
+      "required": [
+        "role"
+      ],
+      "additionalProperties": false
+    },
+    "contacts": {
+      "type": "object",
+      "patternProperties": {
+        "": {
+          "$ref": "#/definitions/contact"
+        }
+      }
+    },
+    "default-registry": {
+      "type": "object",
+      "allOf": [
+        {
+          "oneOf": [
+            {
+              "$ref": "#/definitions/builtin-registry"
+            },
+            {
+              "$ref": "#/definitions/git-registry"
+            },
+            {
+              "$ref": "#/definitions/filesystem-registry"
+            }
+          ]
+        },
+        {
+          "not": {
+            "$ref": "#/definitions/packages-list"
```

Comment:
I'm not 100% sure here, but I think that this "not" will evaluate as success if the user puts an illegal character into one of the port names.

I think a better approach to express this would be to instead list out the full union of allowed field names and leave `"packages"` off (with `"additionalProperties": false`).

Reply:
It will evaluate to false, but none of the things in "oneOf" will evaluate it to true so the overall constraint will still fail.